### ISSUE TITLE
Incorrect output for `new (a::b)()`

### ISSUE THREAD

**Issue by bakkot** [prettier/prettier#6149]
**Prettier 1.17.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEU4HcAEAKAhkpAIwEptiBuAHShABoQIAHGAS2gGdlRcAnHidAAVeCTilwA3CCwAmdEIR64wAazgwAyo2UsoAc2QweAVzj0AFjAC2AGwDq5lvHbawcDaKcsJTgJ7Jwdk56XXY4HhhBJT0rXGQAM1wbMPoAK3YADwAhJVV1DVwrOAAZXTgEpJSQdIyNXT0bOABFYwh4CuSzEG0eMJ4AwlxCXxtoeUYeXRg7WRhzZAAOAAZ6CYgwuyVGAIm4Poly+gBHVvgopjEQXHYAWjQ4GQf5HjgTlheo3Bi4pETO+jCVhYhhMXXY9UaLTa5V+lS6MCGMxkc2QACZ6EZcCwbPUAMIQKyxAJQaCHEDGMIAFSGYj+YQAvvSgA)
```sh
--parser babylon
```

**Input:**
```jsx
new (a::b)();
```

**Output:**
```jsx
new a::b();
```

**Expected behavior:**
```jsx
new (a::b)();
```

---

The thing that prettier is currently outputting is parsed the same as `(new a)::b()`, which is not the same as the input. This also means running it through prettier [again](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEU4HcAEBDJSBGAFAJQDcAOlCADQgQAOMAltAM7KjYBOXE6ACtwTsU2AG4QmAExoh8XbGADWcGAGV6iplADmyGFwCucWgAsYAWwA2AdVNN4rTWDhrhDpmIcBPZOFbstNqscFww-Ao6FtjIAGbYViG0AFasAB4AQgrKqmrYFnAAMtpwcQlJIKlpato6VnAAioYQ8GWJJiCaXCFcfvjY+N5W0LL0XNowNtIwpsgAHAAMtGMQITYK9H5jcD1ipbQAjs3wEQwiINisALRocFJ3slxwR0xPEdhRMUjx7bQhFkx9EYOqxavUmi1St9yh0YAMplIZsgAEy0AzYJhWWoAYQgFmifig0H2IEMIQAKgMRD8QgBfWlAA) gives different output. See also #6147.

**Comment by duailibe** [comment]
I also found this:

**Prettier 1.17.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEB6VACAklADgVxgB0oo4B3DAQySQCMAKASgG4ST0MB5Qg40ihgZVmtRk1YgANCAi4YAS2gBnZKCoAnDRHIAFTQlUoqANwgKAJtJB0NVMAGs4MAMq57CqAHNkMDfjgZAAsYAFsAGwB1IIV4ZXcwOBdDWIUTWIBPZHBlVRlPZTgNGF07L1CqZAAzKnDCmQArZQAPACE7R2cXKlC4ABlPOGra+pAm5pdPL3C4AEV8CHhhusCQdw1CjWy6KjoM8OhrXA1PGEjLGCDkAA4ABhljiELIu1xs47hNkyGZAEcF+ClORGEBUZQAWjIcAs0OsGjg-wU8NKVHKlSQNRWMkKoQUvn8q2UUxm80WQwxI1WMF25wsl2QACYZH4qApwlMAMIQUIVbJQaA-ED4QoAFV2RkxhQAvlKgA)
```sh
--parser babylon
```

**Input:**
```jsx
// Input
new a::b();

// Output
new (a()::b());
```

**Output:**
```jsx
// Input
new a()::b();

// Output
new (a()::b())();

```

cc @sosukesuzuki

**Comment by thorn0** [comment]
@duailibe It's correct formatting, just like `d = new Date;` is formatted as `d = new Date();`.

**Comment by duailibe** [comment]
@thorn0 Yes I got confused with other stuff

**Comment by bakkot** [comment]
Fixed by #6152.
